src/main/java/com/google/api/codegen/discovery/transformer/go/GoSampleTypeNameConverter.java

```diff
@@ -0,0 +1,114 @@
+ */
+package com.google.api.codegen.discovery.transformer.go;
+
+import com.google.api.codegen.discovery.config.TypeInfo;
+import com.google.api.codegen.discovery.transformer.SampleTypeNameConverter;
+import com.google.api.codegen.util.TypeName;
+import com.google.api.codegen.util.TypedValue;
+import com.google.common.collect.ImmutableMap;
+import com.google.protobuf.Field;
+
+public class GoSampleTypeNameConverter implements SampleTypeNameConverter {
+
+  /**
+   * A map from primitive types in proto to Go counterparts.
+   */
+  private static final ImmutableMap<Field.Kind, String> PRIMITIVE_TYPE_MAP =
+      ImmutableMap.<Field.Kind, String>builder()
+          .put(Field.Kind.TYPE_BOOL, "bool")
+          .put(Field.Kind.TYPE_INT32, "int32")
+          .put(Field.Kind.TYPE_INT64, "int64")
+          .put(Field.Kind.TYPE_UINT32, "uint32")
+          .put(Field.Kind.TYPE_UINT64, "uint64")
+          .put(Field.Kind.TYPE_FLOAT, "float32")
+          .put(Field.Kind.TYPE_DOUBLE, "float64")
+          .put(Field.Kind.TYPE_STRING, "string")
+          .put(Field.Kind.TYPE_ENUM, "string")
+          .build();
+
+  /**
+   * A map from primitive types in proto to zero value in Go.
+   */
+  private static final ImmutableMap<Field.Kind, String> PRIMITIVE_ZERO_VALUE =
+      ImmutableMap.<Field.Kind, String>builder()
+          .put(Field.Kind.TYPE_BOOL, "false")
+          .put(Field.Kind.TYPE_INT32, "0")
+          .put(Field.Kind.TYPE_INT64, "0")
+          .put(Field.Kind.TYPE_UINT32, "0")
+          .put(Field.Kind.TYPE_UINT64, "0")
+          .put(Field.Kind.TYPE_FLOAT, "0.0")
+          .put(Field.Kind.TYPE_DOUBLE, "0.0")
+          .put(Field.Kind.TYPE_STRING, "\"\"")
+          .put(Field.Kind.TYPE_ENUM, "\"\"")
+          .build();
+
+  String packagePrefix;
+
+  public GoSampleTypeNameConverter(String packagePrefix) {
+    this.packagePrefix = packagePrefix;
+  }
+
+  @Override
+  public TypeName getServiceTypeName(String apiTypeName) {
+    // N/A
+    return null;
+  }
+
+  @Override
+  public TypeName getRequestTypeName(String apiTypeName, TypeInfo typeInfo) {
+    return getTypeName(typeInfo);
+  }
+
+  @Override
+  public TypeName getTypeName(TypeInfo typeInfo) {
+    if (typeInfo.isArray()) {
+      TypeName elementTypeName = getTypeNameForElementType(typeInfo);
+      return new TypeName("", "", "[]%i{}", elementTypeName);
+    }
+    return getTypeNameForElementType(typeInfo);
+  }
+
+  @Override
+  public TypeName getTypeNameForElementType(TypeInfo typeInfo) {
+    String primitiveTypeName = PRIMITIVE_TYPE_MAP.get(typeInfo.kind());
+    if (primitiveTypeName != null) {
+      return new TypeName(primitiveTypeName);
+    }
+    if (typeInfo.kind() != Field.Kind.TYPE_MESSAGE) {
+      throw new IllegalArgumentException("unsupported type kind: " + typeInfo.kind());
+    }
+    if (typeInfo.isMap()) {
+      throw new IllegalArgumentException("map types are unsupported");
+    }
+    // TODO(garrettjones): Is this okay?
```

Comment:
@garrettjonesgoogle Is this okay? I'm using a static method in the sample namer to pull apart the Go package. Mostly I just want this functionality in one place, since I use it twice.